### PR TITLE
Extract reporters as separate artifacts

### PR DESCRIPTION
**Pull request by shlomisut**
This is a rather big PR - but mostly just moving files around.

Restructuring the project - creating 3 artifacts
1. riemann-java-client 
2. metrics2-riemann-reporter
3. metrics3-riemann-reporter

Reporter modules are dependant on the java client module
This should allow to pick the right artifact for the need, not forcing metrics dependencies as part of the Java client, or mixing metric versions.

Thanks @tarlog for the maven help

### CODE REVIEW

File: src/main/java/com/codahale/metrics/riemann/JvmMetricSet.java [hunk 1]
```diff
@@ -1,118 +0,0 @@
-package com.codahale.metrics.riemann;
```
Comment (aphyr) [inline src/main/java/com/codahale/metrics/riemann/JvmMetricSet.java]: Did you mean to copy this file somewhere, or delete it entirely?

File: riemann-java-client/src/main/java/com/aphyr/riemann/client/TcpHandler.java [hunk 2]
```diff
@@ -1,26 +1,9 @@
 package com.aphyr.riemann.client;
 
-import java.net.InetSocketAddress;
```
Comment (aphyr) [inline riemann-java-client/src/main/java/com/aphyr/riemann/client/TcpHandler.java]: There's a lot of import deletions here--I'm guessing your IDE cleaned up the imports automatically? Can you confirm everything still builds OK? :)

File: src/main/java/com/codahale/metrics/riemann/JvmMetricSet.java (hunk 1 above)
Comment (shlomisut) [inline src/main/java/com/codahale/metrics/riemann/JvmMetricSet.java]: Delete - this is using metrics2 to report in metrics3, which is part of what I want to remove (the duplicate dependency)

File: riemann-java-client/src/main/java/com/aphyr/riemann/client/TcpHandler.java (hunk 2 above)
Comment (shlomisut) [inline riemann-java-client/src/main/java/com/aphyr/riemann/client/TcpHandler.java]: Yes, this is my pre-commit hock in action :) I'm not sure If I tried a build after these changes, though I generally trust intellij on this.
I'll verify & let you know

Comment (shlomisut) [inline riemann-java-client/src/main/java/com/aphyr/riemann/client/TcpHandler.java]: verified all compiles & passes

**Comment by shlomisut** [comment]
@aphyr Can you revisit this ? thanks!

**Comment by shlomisut** [comment]
lets give this another shot... @aphyr  ?

File: src/main/java/com/codahale/metrics/riemann/JvmMetricSet.java (hunk 1 above)
Comment (aphyr) [inline src/main/java/com/codahale/metrics/riemann/JvmMetricSet.java]: Excellent, looks good. :)

**Comment by aphyr** [comment]
Sorry bout the delay, @shlomisut, been backlogged cuz of conference season. Thanks so much for doing this! :)

**Comment by aphyr** [comment]
```
[WARNING] Some problems were encountered while building the effective model for com.aphyr:riemann-java-client:jar:0.4.0
[WARNING] 'build.plugins.plugin.version' for org.apache.felix:maven-bundle-plugin is missing. @ com.aphyr:riemann-java-client-parent:0.4.0, /home/aphyr/riemann-java-client/pom.xml, line 100, column 21
[WARNING] 
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING] 
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
```

I'm a maven newbie--any idea what to do about this?

**Comment by aphyr** [comment]
Also looks like metrics3-riemann-reporter doesn't actually build:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:2.9.1:jar (attach-javadocs) on project metrics3-riemann-reporter: MavenReportException: Error while creating archive:
[ERROR] Exit code: 1 - /home/aphyr/riemann-java-client/metrics3-riemann-reporter/src/main/java/com/codahale/metrics/riemann/RiemannReporter.java:177: error: type arguments not allowed here
[ERROR] * @param tags a {@link Collection<String>}
[ERROR] ^
[ERROR] /home/aphyr/riemann-java-client/metrics3-riemann-reporter/src/main/java/com/codahale/metrics/riemann/RiemannReporter.java:180: warning - Tag @link:illegal character: "60" in "Collection<String>"
[ERROR] /home/aphyr/riemann-java-client/metrics3-riemann-reporter/src/main/java/com/codahale/metrics/riemann/RiemannReporter.java:180: warning - Tag @link:illegal character: "62" in "Collection<String>"
```

**Comment by aphyr** [comment]
I moved the version back to 0.3.2-SNAPSHOT (I prefer to keep stuff on snapshot until we're ready for a release), and pushed versions of the client and metrics2-reporter to clojars. Want to give these a shot? Once all the artifacts build I'll cut a proper release and we'll be good to go. :)

**Comment by shlomisut** [comment]
thanks for looking into this!
About the Maven warning - I have no idea either... never did much in Maven myself (gradle mostly for me) but my sentiment is "If it builds it's OK" 
Which brings me to the second issue - it's doesn't build. :(  From the message you shared it seems to be an issue with the JavaDoc, which I did not validate. I executed ```maven package``` and it built OK "On my machine". 

In short - I'll have a look on both Issues in the next few days & push another PR
Thanks again

**Pull request merged**